overlord/devicestate/firstboot.go

```diff
@@ -216,10 +221,19 @@ func populateStateFromSeedImpl(st *state.State) ([]*state.TaskSet, error) {
 			flags.Required = true
 		}
 
-		ts, _, err := installSeedSnap(st, sn, flags)
+		ts, info, err := installSeedSnap(st, sn, flags)
 		if err != nil {
 			return nil, err
 		}
+		infos = append(infos, info)
+		infoToTs[info] = ts
+	}
+
+	// now add/chain the tasksets in the right order, note that we
+	// only have tasksets that we did not already seeded
```

Comment:
but we need to do something about the base of the gadget though,  we can expect it to be core18 or core but right now with the code as it is here if it's core18 it would still be installed too late

Reply:
Indeed, I added code now that detects if a different base from model is in use and errors

Reply:
I reverted the base detection code again and added a TODO to improve the sorting instead.